scipy/sparse/_sputils.py

```diff
@@ -150,6 +150,94 @@ def getdata(obj, dtype=None, copy=False) -> np.ndarray:
     return data
 
 
+def safely_cast_index_arrays(A, idx_dtype=np.int32, msg=""):
+    """Return cast index arrays of `A` to idx_dtype safely.
```

Comment:
```suggestion
    """Safely cast sparse array indices to `idx_dtype`.
```

I think the subtleties are too much to explain in the first line: "not in place", "returns original if no cast", etc.